docs/pages/application-access/cloud-apis/aws-console.mdx

```diff
@@ -1,186 +1,390 @@
+   run the Teleport Application Service:
+
+   ```json
+   {
+     "Version": "2012-10-17",
+     "Statement": [
+       {
+         "Effect": "Allow",
+         "Principal": {
+           "AWS": "arn:aws:iam::<Var name="AWS_ACCESS_ACCOUNT" />:role/TeleportAWSAccess"
+         },
+         "Action": "sts:AssumeRole"
+       }
+     ]
+   }
+   ```
+
+   In the configuration we show in this guide, the Teleport Application Service
+   assumes the `TeleportAWSAccess` role, then uses that role to assume the
+   `ExampleReadOnlyAccess` role. With the trust policy above, AWS authorizes
+   this operation. (We will create the `TeleportAWSAccess` role later in Step
+   1.)
+
+   <Details title="Cross-account access">
+
+   If you are configuring the Application Service to proxy access to IAM roles
+   in another AWS account, you need to modify the trust policy of the target
+   roles to allow access from the account used by the Teleport Application
+   Service:
+
+   ```json
+   {
+     "Version": "2012-10-17",
+     "Statement": [
+       {
+         "Effect": "Allow",
+         "Principal": {
+           "AWS": "arn:aws:iam::<Var name="account-id"/>:root"
+         },
+         "Action": "sts:AssumeRole"
+       }
+     ]
+   }
+   ```
+
+   This policy allows the Teleport Application Service's AWS account to assume
+   roles in the target account.
+
+   </Details>
+
+1. Run the following commands to create the `ExampleReadOnlyAccess` role:
+
+   ```code
+   $ aws iam create-role --role-name "ExampleReadOnlyAccess" \
+   --assume-role-policy-document file://ro-access.json
+   ```
+
+1. Get the ARN of the AWS-managed `ReadOnlyAccess` policy so you can attach it
+   to your role:
+
+   ```code
+   $ ARN=$(aws iam list-policies --output text --query "Policies[?PolicyName=='ReadOnlyAccess'].Arn")
+   ```
+
+1. Attach the `ReadOnlyAccess` policy to the role:
+
+   ```code
+   $ aws iam attach-role-policy --role-name ExampleReadOnlyAccess --policy-arn $ARN
+   ```
+
+### Give the Teleport Application Service permissions to assume roles
+
+In this section, you will create an IAM role that allows the Teleport
+Application Service to assume other IAM roles in order to proxy user traffic to
+AWS APIs.
+
+1. Define a trust policy to enable the Teleport Application Service to assume
+   the role you will create. 
+
+   <Tabs>
+   <TabItem label="EC2">
+
+   Create a file called `app-service-tp.json`:
+
+   ```json
+   {
+     "Version": "2012-10-17",
+     "Statement": [
+       {
+         "Effect": "Allow", 
+         "Principal": {
+           "Service": "ec2.amazonaws.com"
+         },
+         "Action": "sts:AssumeRole"
+       }
+     ]
+   }
+   ```
+
+   </TabItem>
+   <TabItem label="EKS">
+
+   Retrieve your OIDC issuer ID, assigning <Var name="cluster-name" /> to the name
+   of your EKS cluster and <Var name="eks-region" /> to the AWS region where your
+   EKS cluster is running:
```

Comment:
Should we recommend running `eksctl create iamserviceaccount` with the `--override-existing-serviceaccounts` flag and a `--name` flag pointing to the `teleport-kube-agent` service account? 

One consideration is that, in this guide, we tell the user to install the `teleport-kube-agent` Helm chart (which creates a service account for the Teleport Application Service) after creating the IAM role.

Reply:
> One consideration is that, in this guide, we tell the user to install the teleport-kube-agent Helm chart (which creates a service account for the Teleport Application Service) after creating the IAM role.

I see. If that's the case then it's good. 

To be honest I don't know what ppl do for IAM role for service account in the field. Personally I would still use eksctl since it's easier, then use that created service account to create App service with Helm.

Maybe @hugoShaka @anto have more insight on this.